integration/test/test_power_governor.py

```diff
@@ -34,73 +34,68 @@
 import sys
 import unittest
 import os
+import pandas
 
 import geopm_context
 import geopmpy.io
 
 import util
-
-@util.skip_unless_do_launch()
-class TestIntegration(unittest.TestCase):
-    def setUp(self):
-        self.longMessage = True
-        self._agent = 'power_governor'
-        self._options = {'power_budget': 150}
-        self._tmp_files = []
-        self._output = None
-        self._power_limit = geopm_test_launcher.geopmread("MSR::PKG_POWER_LIMIT:PL1_POWER_LIMIT board 0")
-        self._frequency = geopm_test_launcher.geopmread("MSR::PERF_CTL:FREQ board 0")
-        self._original_freq_map_env = os.environ.get('GEOPM_FREQUENCY_MAP')
-
-    def tearDown(self):
-        geopm_test_launcher.geopmwrite("MSR::PKG_POWER_LIMIT:PL1_POWER_LIMIT board 0 " + str(self._power_limit))
-        geopm_test_launcher.geopmwrite("MSR::PERF_CTL:FREQ board 0 " + str(self._frequency))
-        if self._original_freq_map_env is None:
-            if 'GEOPM_FREQUENCY_MAP' in os.environ:
-                os.environ.pop('GEOPM_FREQUENCY_MAP')
-        else:
-            os.environ['GEOPM_FREQUENCY_MAP'] = self._original_freq_map_env
-
-
-
-
-    # TODO: move to test_power_governor.py
-    @util.skip_unless_run_long_tests()
-    def test_power_consumption(self):
-        name = 'test_power_consumption'
-        report_path = name + '.report'
-        trace_path = name + '.trace'
-        num_node = 4
+import geopm_test_launcher
+
+
+class TestIntegration_power_governor(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """Create launcher, execute benchmark and set up class variables.
+
+        """
+        sys.stdout.write('(' + os.path.basename(__file__).split('.')[0] +
+                         '.' + cls.__name__ + ') ...')
+        test_name = 'test_power_governor'
+        cls._report_path = '{}.report'.format(test_name)
+        cls._trace_path = '{}.trace'.format(test_name)
+        cls._agent_conf_path = test_name + '-agent-config.json'
+        # Clear out exception record for python 2 support
+        geopmpy.error.exc_clear()
+        # Set the job size parameters
+        cls._num_node = 4
         num_rank = 16
         loop_count = 500
-        app_conf = geopmpy.io.BenchConf(name + '_app.config')
-        self._tmp_files.append(app_conf.get_path())
```

Comment:
I see that tmp_files tracking has been removed.  Does this mean these tests will be littering files all over the place?

Reply:
Yes. We had discussed using git to deal with extra files.  

There should also be zero requirement that the tests be run in the git repo now as long as the integration run_config.sh file is sourced.  We can double check that this is true before merging new-profile to dev.

Reply:
I don't think I was involved in the discussion about using git for those files.

The problem I see there is that we had been using tmp_files to conditionally remove the files in the test if the test passed.  If it failed, the files were saved for debug purposes.  That was the reason that this line exists up in setUpClass() for Python2: ```geopmpy.error.exc_clear()```.

If the idea was to invoke ```git clean``` if a test passed, that will remove all the files from any previously failed tests.  I don't see how git can understand what passed vs. failed.  I'm open to discussion if you've already figured this out though.

Reply:
One problem that I don't think we ever resolved is that the check in teardown for `sys.exc_info()` is always `(None, None, None)` so it incorrectly cleans up files in python3 even when the test fails.